test/test_optim.py

```diff
@@ -226,7 +226,9 @@ def test_param_group_with_lrscheduler_goes_right_direction(self, device, dtype,
             bias = Parameter(torch.randn((10), device=device, dtype=dtype))
             inpt = torch.randn(5, device=device, dtype=dtype)
 
-            optimizer = optim_cls([{"params": [weight]}, {"params": [bias], "lr": 0.01}])
+            # avoid endless recompiles by wrapping LR in a tensor if we're compiling
+            lr = torch.tensor(0.01) if torch._utils.is_compiling() else 0.01
```

Comment:
will dynamo eventually auto-do this?

Reply:
If not, then we support LRSched only when optims use a Tensor LR, yes?

Reply:
Yes, we only support LRSched when optims use a Tensor LR. I'm on the fence on whether to do this wrapping automatically. We can but on some optims there are more kernels generated (reduced performance) so I think I won't for now. If this turns into a large issue it's pretty easy to do and a lighter option could also be to add a warning on excessive recompiles due to LR changing.

Reply:
I think not doing autowrapping is the right thing for now--with a warning that it is recompiling a ton when the lr is not a Tensor.